### DISCUSSION TITLE
Help understanding this code

### DISCUSSION THREAD

**Discussion by designermonkey** [thephpleague/commonmark#661] [Q&A] [answered]
I need to make an extension that allows me to parse Mustache templating as referenced links.

I just cannot figure out how this all works, especially for referenced links in the markdown spec.

```
[id] /link/here "optional title"
```

Can anyone explain how this is parsed? I need to parse the following

```
[id] {{ link.url }} "{{ link.title }}"
```

so that the output in html is

```
<a href="{{ link.url }}" title="{{ link.title }}">...</a>
```

**Comment by colinodell** [comment, marked as answer]
Sorry for the (extremely long) delay in answering this.

References are parsed using the `OpenBrackerParser` and `CloseBracketParser` - see https://github.com/thephpleague/commonmark/blob/master/src/Inline/Parser/CloseBracketParser.php#L187 for how we check whether there's a reference.

I'm going to close this out since it's been open for a while, but I hope this gives you a good starting point!
